### PR TITLE
fix: assets permission issues #1926

### PR DESCRIPTION
**Pull request by regevbr**
fix: #1926
Added permission filtering when listing assets and folders

### CODE REVIEW

**Comment by regevbr** [comment]
@NGPixel did you have a chance to look at this PR? I think this is an important fix...

**Comment by regevbr** [comment]
@NGPixel sorry to bother, I'm sure you are very busy. I saw you made a new release which is aswome and I already upgraded to it. But in order for me to publicly expose parts of my wiki, I need that fix. I hope you can CR it before the next release.

File: server/graph/resolvers/asset.js [hunk 1]
```diff
@@ -20,18 +20,29 @@ module.exports = {
       if (args.kind !== 'ALL') {
         cond.kind = args.kind.toLowerCase()
       }
-      const result = await WIKI.models.assets.query().where(cond)
-      return result.map(a => ({
+      const folderHierarchy = await WIKI.models.assetFolders.getHierarchy(args.folderId)
+      const folderPath = folderHierarchy.map(h => h.slug).join('/')
+      let results = await WIKI.models.assets.query().where(cond)
+      results = _.filter(results, r => {
```
Comment (NGPixel) [inline server/graph/resolvers/asset.js:26]: Don't reassign `results` here. Should be `const results = ...` then you return the `_filter().map()` as 1 chained statement.

File: server/graph/resolvers/asset.js [hunk 2]
```diff
@@ -20,18 +20,29 @@ module.exports = {
...
     async folders(obj, args, context) {
-      const result = await WIKI.models.assetFolders.query().where({
+      let results = await WIKI.models.assetFolders.query().where({
         parentId: args.parentFolderId === 0 ? null : args.parentFolderId
       })
-      // TODO: Filter by page rules
-      return result
+      const parentHierarchy = await WIKI.models.assetFolders.getHierarchy(args.parentFolderId)
+      const parentPath = parentHierarchy.map(h => h.slug).join('/')
+      results = _.filter(results, r => {
```
Comment (NGPixel) [inline server/graph/resolvers/asset.js:41]: Same issue here. Don't reassign, return the result directly.

**Comment by regevbr** [comment]
Thanks @NGPixel I fixed your comments.

**Comment by regevbr** [comment]
Thanks @NGPixel! Really an awesome project :-)

**Pull request merged**